File: python/n-008.py
# -*- coding: utf-8 -*-


def cipher(s):
    def _cipher(c):
        o = ord(c)
        if o >= 97 and o <= 122:
            return chr(219-o)
        else:
            return c
    return ''.join([_cipher(c) for c in s])

if __name__ == '__main__':
    song = '99 Bottles of Beer.'
    if not song == cipher(cipher(song)):
        print('ng')

File: ruby/n100/lib/n008.rb
class String
  def cipher
    ordinals = self.chars.map(&:ord)
    cs = ordinals.map do |o|
      if o >= 97 && o <= 122
        (219 - o).chr
      else
        o.chr
      end
    end
    cs.join
  end
end
